cmd/root.go

```diff
@@ -335,7 +325,28 @@ func mergeEnvs(envs []fn.Env, envToUpdate *util.OrderedMap, envToRemove []string
 
 // setPathFlag ensures common text/wording when the --path flag is used
 func setPathFlag(cmd *cobra.Command) {
-	cmd.Flags().StringP("path", "p", cwd(), "Path to the project directory (Env: $FUNC_PATH)")
+	cmd.Flags().StringP("path", "p", ".", "Path to the project directory (Env: $FUNC_PATH)")
+}
+
+// getPathFlag returns the value of the --path flag.
+// The special value '.' is returned as the abolute path to the current
+// working directory.
+func getPathFlag() string {
+	path := viper.GetString("path")
+	if path == "." {
+		path = cwd()
+	}
+	return path
+}
+
+// cwd returns the current working directory or exits 1 printing the error.
+func cwd() (cwd string) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to determine current working directory: %v", err)
```

Comment:
Wouldn't be `panic()` better?

Reply:
This is not just like my preference.
With `os.Exit()` the process dies instantly and no `defer`s are not invoked.

Reply:
`os.Exit()` should be only called from `main()`

Reply:
I am open to that, yes.  This is currently set as nonfatal because I am not convinced being unable to retrieve the current working directory is a problem that is fatal.  As you mentioned the default is to presume CWD.

If this inability to determine current working directory is a fatal error, then we could panic, yes, or change this to be a standard error-returning function which would fal upwards to the root-level catch which exits cleanly with (potentially) a custom numerical code for this particular exit condition.